optuna/study/_frozen.py

```diff
@@ -40,7 +40,7 @@ def __init__(
         study_name: str,
         direction: Optional[StudyDirection],
         user_attrs: Dict[str, Any],
-        system_attrs: Dict[str, Any],
+        system_attrs: Dict[str, Any],  # TODO(gen740): Change Any to JSONSerializable
```

Comment:
This is my personal opinion, but the TODO comment to leave in the code should be self-explanatory as to whether or not it should be done. Whether or not to make the type JsonSerializable for an API that is open to users is a matter of debate and needs to be agreed upon by the developers. Wouldn't it be enough, for example, to leave such a TODO discussion on the PR like this now?